### PR TITLE
Fix/1632/reset street layout global config

### LINKED ISSUE #1624: The "maximum treemap files" is always visible
# Bug

## Expected Behavior

GIVEN a user has opened the global settings

WHEN not using the "TreeMapStreet" layout

THEN there should not be an option to specify the maximum treemap files as it has no functionality on those layouts.

## Actual Behavior

The option is always visible.

## Screenshots

image

## Specifications
- is released in [online-demo](https://maibornwolff.github.io/codecharta/visualization/app/index.html?file=codecharta.cc.json&file=codecharta_analysis.cc.json): yes
- CodeCharta Version: 1.65.0

### LINKED ISSUE #1632: reset street view global config not working
# Bug

## Expected Behavior

GIVEN a user has opened the global config
WHEN modifying the layout or the street layout maximum treemap files, then resetting the global configuration
THEN layout and max treemap files options should revert to default values.

## Actual Behavior

The options does not revert to default values.

## Screenshots

image

## Specifications
- is released in [online-demo](https://maibornwolff.github.io/codecharta/visualization/app/index.html?file=codecharta.cc.json&file=codecharta_analysis.cc.json): yes 
- CodeCharta Version: 1.65.0

### PR DESCRIPTION
**Pull request by Med-H**
# Reset layout option and street map max files in global settings

closes #1632 
closes #1624

## Description

GIVEN a user has opened the global config
WHEN modifying the layout or the street layout maximum treemap files, then resetting the global configuration
THEN layout and max treemap files options should revert to default values.

thuis PR closes this issue.

### CODE REVIEW

**Review by Med-H — COMMENTED** [review]
just modified the comment

**Comment by BridgeAR** [comment]
This does not yet fix our flaky e2e tests:

```
FAIL app/codeCharta/e2e/url.e2e.ts (44.26 s)
  ● [secure] › should throw errors when file parameters in url are invalid and load sample data instead
    TimeoutError: waiting for selector `.md-select-menu-container.md-active > md-select-menu` failed: timeout 30000ms exceeded
      12 | 		await this.clickChooser()
      13 | 
    > 14 | 		await page.waitForSelector(".md-select-menu-container.md-active > md-select-menu")
         | 		           ^
      15 | 		const content = await page.$eval(".md-select-menu-container.md-active > md-select-menu", element => element["innerText"])
      16 | 		return content.split("\n")
      17 | 	}
```

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,2104 +1,2108 @@
...
+
+### Changed
+
+### Removed 🗑
+
+### Fixed 🐞
+
+-   Global settings not reverting to default ones ([#1632](https://github.com/MaibornWolff/codecharta/issues/1632))
+-   Maximum treemap files shown in squarified node ([#1624](https://github.com/MaibornWolff/codecharta/issues/1624))
+-   Fixed kotlin test ([#1322](https://github.com/MaibornWolff/codecharta/issues/1322))
```
Comment (BridgeAR) [inline CHANGELOG.md:20]: This part should not be relevant for our customers. Therefore it's something we could in theory mention in "Chore" but I prefer not to add it at all.

```suggestion
```

File: analysis/import/SCMLogParser/src/main/kotlin/de/maibornwolff/codecharta/importer/scmlogparser/input/metrics/CalendarWeek.kt [hunk 2]
```diff
@@ -36,8 +45,8 @@ internal data class CalendarWeek(private val week: Int, private val year: Int) :
             return dateTime.dayOfYear < 7
         }
 
-        private fun isFirstOrSecondWeek(kalenderWeeknWeek: Int): Boolean {
-            return kalenderWeeknWeek <= 2
+        private fun isFirstOrSecondWeek(calendarWeek: Int): Boolean {
+            return calendarWeek <= 2
```
Comment (BridgeAR) [inline analysis/import/SCMLogParser/src/main/kotlin/de/maibornwolff/codecharta/importer/scmlogparser/input/metrics/CalendarWeek.kt:49]: ❤️

File: analysis/import/SCMLogParser/src/main/kotlin/de/maibornwolff/codecharta/importer/scmlogparser/input/metrics/CalendarWeek.kt [hunk 3]
```diff
@@ -19,7 +20,15 @@ internal data class CalendarWeek(private val week: Int, private val year: Int) :
...
-            return b.week + 52 * b.year - (a.week + 52 * a.year)
+            return ChronoUnit.WEEKS.between(getWeekDate(a.year, a.week), getWeekDate(b.year, b.week)).toInt()
+        }
+
+        private fun getWeekDate(year: Int, week: Int): OffsetDateTime? {
+            return OffsetDateTime.now().withYear(year)
+                .with(WeekFields.ISO.weekOfWeekBasedYear(), week.toLong())
+                .with(WeekFields.ISO.dayOfWeek(), 1)
+                .withHour(12).withMinute(0).withSecond(0).withNano(0)
+                .withOffsetSameInstant(ZoneOffset.UTC)
```
Comment (BridgeAR) [inline analysis/import/SCMLogParser/src/main/kotlin/de/maibornwolff/codecharta/importer/scmlogparser/input/metrics/CalendarWeek.kt:31]: This is not immediately obvious what this does. I suggest to add a short comment what this tries to solve :-)

File: visualization/app/codeCharta/ui/codeMap/codeMap.label.service.ts [hunk 4]
```diff
@@ -117,6 +117,7 @@ export class CodeMapLabelService implements CameraChangeSubscriber {
 		}
 	}
 
+	// TODO Add a test to ensure that scaling is not additive
```
Comment (BridgeAR) [inline visualization/app/codeCharta/ui/codeMap/codeMap.label.service.ts:120]: This seems to be unrelated?

**Review by BridgeAR — COMMENTED** [review]
I am a bit surprised that we reformat our changelog all the time. How could we guarantee this does not happen anymore? :-)

File: analysis/import/SCMLogParser/src/main/kotlin/de/maibornwolff/codecharta/importer/scmlogparser/input/metrics/CalendarWeek.kt [hunk 5]
```diff
@@ -19,7 +20,16 @@ internal data class CalendarWeek(private val week: Int, private val year: Int) :
         }
 
         fun numberOfWeeksBetween(a: CalendarWeek, b: CalendarWeek): Int {
-            return b.week + 52 * b.year - (a.week + 52 * a.year)
+            return ChronoUnit.WEEKS.between(getWeekDate(a.year, a.week), getWeekDate(b.year, b.week)).toInt()
+        }
+
+        private fun getWeekDate(year: Int, week: Int): OffsetDateTime? {
+            // returns the date of Monday based on the week and year
```
Comment (shaman-apprentice) [inline analysis/import/SCMLogParser/src/main/kotlin/de/maibornwolff/codecharta/importer/scmlogparser/input/metrics/CalendarWeek.kt:27]: [Optional] I like to have longer variable names like `getDateOfWeeksMonday` instead of comments, which tend to rotten 😇 Guess that would also meet @BridgeAR wish for a comment.

File: visualization/app/puppeteer.helper.ts [hunk 6]
```diff
@@ -1,8 +1,12 @@
 import path from "path"
+import { setDefaultOptions } from 'expect-puppeteer'
 import { ConsoleMessage } from "puppeteer"
 
 export const CC_URL = `file:${path.join(__dirname, "../dist/webpack/index.html")}`
-
+// expect-puppeteer toClick timeout does not work it might be the reason of flaky tests,
+// changed default options globally
+// [toClick issue](https://github.com/smooth-code/jest-puppeteer/issues/202)
+setDefaultOptions({ timeout: 6000 })
```
Comment (shaman-apprentice) [inline visualization/app/puppeteer.helper.ts:9]: I have always a bad feeling with huge default timeouts 😟 But if it is necessary this shouldn't be a blocker

**Review by shaman-apprentice — APPROVED** [review]
